File: MakeItRain/app/src/main/java/org/techtown/makeitrain/MainActivity.java
package org.techtown.makeitrain;

import androidx.appcompat.app.AppCompatActivity;
import androidx.databinding.DataBindingUtil;

import android.content.Context;
import android.icu.text.NumberFormat;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.inputmethod.InputMethodManager;
import android.widget.Button;
import android.widget.EditText;
import android.widget.TextView;

import com.google.android.material.snackbar.Snackbar;

import org.techtown.makeitrain.data.Bio;
import org.techtown.makeitrain.databinding.ActivityMainBinding;

public class MainActivity extends AppCompatActivity {
    private ActivityMainBinding binding;
    private final Bio bio = new Bio();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        binding = DataBindingUtil.setContentView(this, R.layout.activity_main);
        bio.setName("baesumin");

        binding.setBio(bio);
        binding.doneButton.setOnClickListener(view -> addHobbies(view));

    }

    public void addHobbies(View view) {
        bio.setHobbies(String.format("Hobbies:%s ", binding.enterHobbies.getText().toString().trim()));
        binding.invalidateAll();
        binding.hobbiesText.setVisibility(View.VISIBLE);

        InputMethodManager inputMethodManager = (InputMethodManager) getSystemService(Context.INPUT_METHOD_SERVICE);
        inputMethodManager.hideSoftInputFromWindow(view.getWindowToken(),0);
    }
}

File: WorkManagerDemo1/app/src/main/java/org/techtown/workmanagerdemo1/MainActivity.kt
package org.techtown.workmanagerdemo1

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.TextView
import androidx.lifecycle.Observer
import androidx.work.OneTimeWorkRequest
import androidx.work.WorkManager

class MainActivity : AppCompatActivity() {
  override fun onCreate(savedInstanceState: Bundle?) {
    super.onCreate(savedInstanceState)
    setContentView(R.layout.activity_main)
    findViewById<Button>(R.id.button).setOnClickListener {
      setOneTimeWorkRequest()
    }
  }

  private fun setOneTimeWorkRequest(){
    val workManager = WorkManager.getInstance(applicationContext)
    val uploadRequest = OneTimeWorkRequest.Builder(UploadWorker::class.java)
      .build()
    workManager.enqueue(uploadRequest)
    workManager.getWorkInfoByIdLiveData(uploadRequest.id)
      .observe(this, Observer {
        findViewById<TextView>(R.id.textView).text = it.state.name
      })
  }
}